src/wp-includes/rest-api/endpoints/class-wp-rest-templates-controller.php

```diff
@@ -215,6 +215,23 @@ public function get_items( $request ) {
 	 * @return true|WP_Error True if the request has read access for the item, WP_Error object otherwise.
 	 */
 	public function get_item_permissions_check( $request ) {
+		$template = $this->get_block_template( $request['id'] );
+		if ( is_wp_error( $template ) ) {
+			return $template;
+		}
+
+		if ( ! $template->wp_id ) {
+			return $this->permissions_check( $request );
+		}
+
+		if ( current_user_can( 'read_post', $template->wp_id ) ) {
+			return true;
+		}
+
+		if ( 'edit' === $request['context'] && current_user_can( 'edit_post', $template->wp_id ) ) {
+			return true;
+		}
```

Comment:
Shouldn't this happen before the `read_post` check?

Reply:
We'd also want this to fail if they request an `edit` context but can't edit it.